stream.go

```diff
@@ -0,0 +1,126 @@
+package kinesis
+
+import (
+	"fmt"
+	"log"
+	"text/template"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/gliderlabs/logspout/router"
+)
+
+type ErrStreamNotReady struct {
+	s string
+}
+
+func (e *ErrStreamNotReady) Error() string {
+	return fmt.Sprintf("not ready, stream: %s", e.s)
+}
+
+// Stream represents a stream that will send messages to its writer.
+type Stream struct {
+	client     Client
+	name       string
+	tags       *map[string]*string
+	Writer     *writer
```

Comment:
Also minor, should just make this unexported. We probably wouldn't want consumers to use this directly.